rdkit/Chem/Draw/UnitTestDraw.py

```diff
@@ -8,185 +7,154 @@
+except ImportError:
+  qtCanvas = None
 
 
 class TestCase(unittest.TestCase):
+  showAllImages = False
 
   def setUp(self):
     if IPythonConsole is not None and Draw.MolsToGridImage == IPythonConsole.ShowMols:
       IPythonConsole.UninstallIPythonRenderer()
     self.mol = Chem.MolFromSmiles('c1c(C[15NH3+])ccnc1[C@](Cl)(Br)[C@](Cl)(Br)F')
 
-  def testCairoFile(self):
+  def _testMolToFile(self):
+    _, fn = tempfile.mkstemp(suffix='.png')
     try:
-      from rdkit.Chem.Draw.cairoCanvas import Canvas
-    except ImportError:
-      logger.info("Skipping cairo test")
-      return
-    os.environ['RDKIT_CANVAS'] = 'cairo'
-
-    foo, fn = tempfile.mkstemp(suffix='.png')
-    foo = None
-    self.assertEqual(os.path.getsize(fn), 0)
+      self.assertEqual(os.path.getsize(fn), 0)
+      Draw.MolToFile(self.mol, fn)
+      self.assertNotEqual(os.path.getsize(fn), 0)
+    finally:
+      os.remove(fn)
 
-    Draw.MolToFile(self.mol, fn)
-
-    self.assertNotEqual(os.path.getsize(fn), 0)
-    try:
-      os.unlink(fn)
-    except Exception:
-      pass
+  @unittest.skipIf(cairoCanvas is None, 'Skipping cairo test')
+  def testCairoFile(self):
+    os.environ['RDKIT_CANVAS'] = 'cairo'
+    self._testMolToFile()
 
+  @unittest.skipIf(aggCanvas is None, 'Skipping agg test')
   def testAggFile(self):
-    try:
-      from rdkit.Chem.Draw.aggCanvas import Canvas
-    except ImportError:
-      logger.info("Skipping agg test")
-      return
     os.environ['RDKIT_CANVAS'] = 'agg'
+    self._testMolToFile()
 
-    foo, fn = tempfile.mkstemp(suffix='.png')
-    foo = None
-    self.assertEqual(os.path.getsize(fn), 0)
-
-    Draw.MolToFile(self.mol, fn)
-
-    self.assertNotEqual(os.path.getsize(fn), 0)
-    try:
-      os.unlink(fn)
-    except Exception:
-      pass
-
+  @unittest.skipIf(spingCanvas is None, 'Skipping sping test')
   def testSpingFile(self):
-    try:
-      from rdkit.Chem.Draw.spingCanvas import Canvas
-    except ImportError:
-      logger.info("Skipping sping test")
-      return
     os.environ['RDKIT_CANVAS'] = 'sping'
+    self._testMolToFile()
 
-    foo, fn = tempfile.mkstemp(suffix='.png')
-    foo = None
-    self.assertEqual(os.path.getsize(fn), 0)
-
-    Draw.MolToFile(self.mol, fn)
-
-    self.assertNotEqual(os.path.getsize(fn), 0)
-    try:
-      os.unlink(fn)
-    except Exception:
-      pass
-
-  def testCairoImage(self):
-    try:
-      from rdkit.Chem.Draw.cairoCanvas import Canvas
-    except ImportError:
-      return
-    os.environ['RDKIT_CANVAS'] = 'cairo'
-
-    img = Draw.MolToImage(self.mol, size=(300, 300))
+  def _testMolToImage(self, mol=None, kekulize=True, options=None, showImage=False, **kwargs):
+    mol = mol or self.mol
+    img = Draw.MolToImage(mol, size=(300, 300), kekulize=kekulize, options=options, **kwargs)
     self.assertTrue(img)
     self.assertEqual(img.size[0], 300)
     self.assertEqual(img.size[1], 300)
+    if self.showAllImages or showImage:
+      img.show()
```

Comment:
Yes. it will show the created images. Was useful during refactoring. Default is off, however to be safe, I'll add a test that ensures showAllImages is False to avoid accidentally merging it into the repository.